src/workerd/jsg/filesystem.h

```diff
@@ -0,0 +1,244 @@
+#pragma once
+
+#include "jsg.h"
+
+namespace workerd::jsg::fs {
+
+// TODO(node-fs): Currently, all files and directories use a fixed last
+// modified time set to the Unix epoch. This is temporary.
+
+// Represents a file system node (file or directory) in the virtual file system.
+// This is an intentionally simplified alternative to the kj::FsNode interface
+// that minimizes it to only the types, methods, and metadata that are needed
+// for this jsg::VirtualFileSystem API.
+class Node: public kj::Refcounted {
+ public:
+  using Type = Node_Type;
+
+  Node() = default;
+  KJ_DISALLOW_COPY_AND_MOVE(Node);
+  virtual ~Node() noexcept(false) = default;
+
+  // Metadata about this filesystem node
+  struct Stat final {
+    Type type = Type::FILE;
+
+    // The size of the node in bytes. For directories, this value will
+    // always be 0.
+    size_t size = 0;
+
+    // The last modified time of the node.
+    kj::Date lastModified = kj::UNIX_EPOCH;
+
+    // The creation time of the node.
+    kj::Date created = kj::UNIX_EPOCH;
+
+    // Indicates if the node is writable.
+    bool writable = false;
+  };
+
+  // Attempt to set the last modified time of this node. If the node is
+  // read-only, this will be a non-op.
+  virtual void setLastModified(Lock& js, kj::Date date = kj::UNIX_EPOCH) = 0;
```

Comment:
Nit: Doesn't feel like this should have a default argument

Reply:
Agree